src/ucp/core/ucp_ep.c

```diff
@@ -3241,27 +3241,23 @@ void ucp_ep_invoke_err_cb(ucp_ep_h ep, ucs_status_t status)
     ucp_ep_ext_control(ep)->err_cb(ucp_ep_ext_gen(ep)->user_data, ep, status);
 }
 
-static UCS_F_ALWAYS_INLINE int
-ucp_ep_is_am_keepalive(ucp_ep_h ep, ucp_rsc_index_t rsc_idx)
+int ucp_ep_is_am_keepalive(ucp_ep_h ep, ucp_rsc_index_t rsc_idx, int is_p2p)
 {
     ucp_worker_iface_t *wiface;
 
-    if (!(ep->flags & UCP_EP_FLAG_REMOTE_ID) ||
+    if (is_p2p || !(ep->flags & UCP_EP_FLAG_REMOTE_ID) ||
```

Comment:
1. rename rsc_idx ro rsc_indes
2. since we already pass rsc_index, i think is_p2p can be calculated from it: 
     `(rsc_index != NULL_RESOURCE) && ucp_ep_config_connect_p2p(worker, &ucp_ep_config(ep)->key, rsc_index)`

```c
int ucp_ep_is_am_keepalive(ucp_ep_h ep, ucp_rsc_index_t rsc_index)
{
    ucp_worker_h worker = ep->worker;

    return /* Not a CM lane */
           (rsc_index != NULL_RESOURCE) && 
           /* Have a remote endpoint ID to send in in the keepalive active message */
           (ep->flags & UCP_EP_FLAG_REMOTE_ID) &&
           /* Transport supports active messages */
           (ucp_worker_iface(worker, rsc_index)->flags & UCT_IFACE_FLAG_AM_BCOPY) &&
           /* Transport is not connected as point-to-point */
           !ucp_ep_config_connect_p2p(worker, &ucp_ep_config(ep)->key, rsc_index);
}
```

Reply:
done

> since we already pass rsc_index, i think is_p2p can be calculated from it:
(rsc_index != NULL_RESOURCE) && ucp_ep_config_connect_p2p(worker, &ucp_ep_config(ep)->key, rsc_index)

we can't use such logic for AUX_EP

Reply:
> we can't use such logic for AUX_EP

because it could be created as to-iface inside wireup_ep and not set from existing lane?

Reply:
>>we can't use such logic for AUX_EP

>because it could be created as to-iface inside wireup_ep and not set from existing lane?

it could be:
- created as to-IFACE inside WIREUP_EP if next_ep is CONNECT_TO_EP.
- set as to-IFACE/to-EP during connection inersecting during CM connection establishment.